### DISCUSSION TITLE
Search Indexes: primary, secondary or none?

### DISCUSSION THREAD

**Discussion by gvlx** [Budibase/budibase#3333] [Help] [answered]
Hi,

What is the purpose of Search Indexes and which type of index should I choose in my columns?

Thank you.

**Comment by mjashanks** [comment, marked as answer]
Hey Gerardo. 

The search index should be used to enhance performance, when you have a large amount of rows in your table. 

So, in your `Design` section (on a `Data Provider`), if you are filtering by a certain column and you have a lot of data, you should add an index to that column. It will make the search faster.

**Comment by gvlx** [comment]
Fine, but what are the differences between primary and secondary indexes?
And are the implications of having several fields in the primary or secondary indexes?
